deployments/apiv2/services/users/src/main.py

```diff
@@ -327,3 +340,78 @@ async def register(
         raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+@app.get("/users")
+async def get_users(request: Request, token=Depends(ProtectedAny(scope=["users:admin"]))):
+    """Get all the information for all the users connected to the admin account that started this request.
+    By default it will be sorted with all active users showing up first, then all the deactivated users
+    """
+    try:
+        async with request.state.pgpool.acquire() as con:
+            list_of_users = []
+            customer_id = uuid.UUID(hex=token["userid"])
+            active_rows = await con.fetch(
+                "SELECT name, email, created_at, last_login,suspended FROM users WHERE customer_id=$1 AND deleted_at IS NULL AND suspended='f'",
+                customer_id,
+            )
+            for row in active_rows:
+                list_of_users.append(
+                    {
+                        "name": row.get("name", ""),
+                        "email": row.get("email", ""),
+                        "date_created": row.get("created_at", ""),
+                        "last_loggedin": row.get("last_login"),
+                        "deactivated": row.get("suspended"),
+                    }
+                )
+            deactive_rows = await con.fetch(
+                "SELECT name, email, created_at, last_login,suspended FROM users WHERE customer_id=$1 AND deleted_at IS NULL AND suspended='t'",
+                customer_id,
+            )
+            for row in deactive_rows:
+                list_of_users.append(
+                    {
+                        "name": row.get("name", ""),
+                        "email": row.get("email", ""),
+                        "date_created": row.get("created_at", ""),
+                        "last_loggedin": row.get("last_login"),
+                        "deactivated": row.get("suspended"),
+                    }
+                )
+            return list_of_users
+    except Exception as e:
+        logger.exception(f"users GET: Unexpected error {repr(e)}")
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+@app.delete("/user-actions/{user_email_to_delete}")
+async def delete_user(
+    request: Request, user_email_to_delete: EmailStr, token=Depends(ProtectedAny(scope=["users:admin"]))
+):
+    """Mark a users selected as deleted.When deleted the user will not be fetched when displaying table of users"""
+    try:
+        async with request.state.pgpool.acquire() as con:
+            update_query = "UPDATE users SET deleted_at=$1  WHERE email=$2"
+            await con.execute(update_query, datetime.datetime.now(), user_email_to_delete)
+    except Exception as e:
+        logger.exception(f"user-actions DELETE: Unexpected error {repr(e)}")
+        raise HTTPException(status_code=status.HTTP_500_INTERNAL_SERVER_ERROR)
+
+
+@app.put("/user-actions/{user_email_to_deactivate}")
```

Comment:
Is there any reason we couldn't make this `@app.put("/users/{email}")`? I'm guessing that down the road we will want to make this a very general purpose route that can edit info about users, and to keep it more RESTful we could use this as the route name instead, as well as the other change I mentioned so that we have GET, PUT, and DELETE `/users/...`. Also if this route will end up being a general purpose route to edit users, we should also name the function something like `edit_user_info` and change the docstring